File: homeloanmanagement/src/main/java/com/cjc/homeloanmanagement/app/model/Customer.java
package com.cjc.homeloanmanagement.app.model;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class Customer {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer customerId;
	private String customerName;
	private String customerDateOfBirth;
	private Integer customerAge;
	private String customerGender;
	private String customerEmail;
	private Long customerMobileNumber;
	private Long customerAdditionalMobileNumber;
	private Double customerTotalLoanRequired;
	@OneToOne(cascade = CascadeType.ALL)
	private AllPersonalDocuments allpersonaldocs;
	@OneToOne(cascade = CascadeType.ALL)
    private CustomerAddress customerAddress;
	@OneToOne(cascade = CascadeType.ALL)
    private MortgageDetails mortgageDetails;
	@OneToOne(cascade = CascadeType.ALL)
	private Profession profession;
	@OneToOne(cascade = CascadeType.ALL)
	private BankAccountDetails bankAccountDetails;
	@OneToOne(cascade = CascadeType.ALL)
	private PropertyInfo propertyinfo;
	@OneToOne(cascade = CascadeType.ALL)
	private GuarantorDetails guarantordetails;
	@OneToOne(cascade = CascadeType.ALL)
	private LoanDisbursement loandisbursement;
	@OneToOne(cascade = CascadeType.ALL)
	private Ledger ledger;
	@OneToOne(cascade = CascadeType.ALL)
	private SanctionLetter sanctionletter;
	@OneToOne(cascade = CascadeType.ALL)
	private CustomerVerification customerverification;




	



    
    

}

File: Front End/HomeLoanManagementProject/src/app/model/customer.ts
import { BankAccountDetails } from "./bank-account-details";
import { CustomerAddress } from "./customer-address";
import { CustomerVerification } from "./customer-verification";
import { DependentInfo } from "./dependent-info";
import { GuarantorDetails } from "./guarantor-details";
import { LoanDisbursement } from "./loan-disbursement";
import { MortgageDetails } from "./mortgage-details";
import { Profession } from "./profession";
import { PropertyInfo } from "./property-info";
import { SanctionLetter } from "./sanction-letter";

export class Customer {
customerId:number;
customerName:string;
customerDateOfBirth:string;
customerAge:number;
customerGender:string;
customerEmail:string;
customerMobileNumber:number;
customerAdditionalMobileNumber:number;
customerTotalLoanRequired:number;
customerAddress:CustomerAddress;
mortgageDetails:MortgageDetails;
profession:Profession;
bankAccountDetails:BankAccountDetails;
propertyinfo:PropertyInfo;
guarantordetails:GuarantorDetails;
loandisbursement:LoanDisbursement;
sanctionletter: SanctionLetter;
customerverification:CustomerVerification;
dependentInfo:DependentInfo;

}
